domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -3,11 +3,109 @@ package dagtraversalmanager_test
+		}
+		defer tearDown(false)
+
+		// "checkExpectedBlockChainDag" - compare BlockAtDepth to a loop on the SelectedChainIterator on a single chain DAG.
+		checkExpectedBlockChainDag := func(highHash *externalapi.DomainHash, depth uint64) {
+			currentBlock := highHash
+			currentBlockData, err := tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+			if err != nil {
+				t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+			}
+			for i := uint64(0); i <= depth; i++ {
+				if currentBlockData.SelectedParent() == nil {
+					break
+				}
+				currentBlock = currentBlockData.SelectedParent()
+				currentBlockData, err = tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+				if err != nil {
+					t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+				}
+			}
+			expected := currentBlock
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(highHash, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+
+			if !blockHash.Equal(expected) {
+				t.Fatalf("Expected block %s but got %s", expected, blockHash)
+			}
+		}
+
+		// "checkExpectedBlockTwoChildrenSameSelectedParent" checks blockAtDepth on 2 children with the same selectedParent.
+		checkExpectedBlockTwoChildrenSameSelectedParent := func(firstChild, secondChild *externalapi.DomainHash, depth uint64) {
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(firstChild, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			expectedSameHash, err := tc.DAGTraversalManager().BlockAtDepth(secondChild, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			if !blockHash.Equal(expectedSameHash) {
+				t.Fatalf("Expected block %s but got %s", expectedSameHash, blockHash)
+			}
+		}
+
+		// "checkExpectedBlockTwoChildrenSameSelectedParent" checks on different chains, on the same DAG, that they merge at the correct point.
+		checkExpectedBlockOnTwoDiffChain := func(firstChild, secondChild *externalapi.DomainHash, firstDepth, secondDepth uint64) {
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(firstChild, firstDepth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			expectedSameHash, err := tc.DAGTraversalManager().BlockAtDepth(secondChild, secondDepth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			if !blockHash.Equal(expectedSameHash) {
+				t.Fatalf("Expected block %s but got %s", expectedSameHash, blockHash)
+			}
+			expectedDiffHash, err := tc.DAGTraversalManager().BlockAtDepth(secondChild, secondDepth-1)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			if blockHash.Equal(expectedDiffHash) {
+				t.Fatalf("Expected to differente block")
+			}
+
+		}
+
+		highHash, err := createAChainDag(params.GenesisHash, tc)
+		if err != nil {
+			t.Fatalf("createAChainDagInBlockAtDepth: %+v", err)
+		}
+		checkExpectedBlockChainDag(highHash, 2)
+
+		firstChild, secondChild, err := expendDagToTwoChildrenSameSelectedParent(highHash, tc)
```

Comment:
Another reason to split this to 3 different tests - you don't really want to "Expend" on the DAG from one test-case to the next, as those should be kept separate.
There's some probability that a change in one test will inadvertently effect some other test.

Reply:
Changed, thanks.